src/Native/System.Native/pal_networking.cpp

```diff
@@ -2013,3 +2018,21 @@ extern "C" Error WaitForSocketEvents(int32_t port, SocketEvent* buffer, int32_t*
 
     return WaitForSocketEventsInner(port, buffer, count);
 }
+
+extern "C" int32_t PlatformSupportsMultipleConnectAttempts()
+{
+#if defined(PLATFORM_SUPPORTS_MULTIPLE_CONNECT_ATTEMPTS)
```

Comment:
Maybe better to use:

```C
#ifdef PLATFORM_SUPPORTS_MULTIPLE_CONNECT_ATTEMPTS
```

here. Generally, `#ifdef` is used, if there is only one macro to check, otherwise `#if defined(X) && defined(Y) || defined(Z)` or `#if SOME_ZERO_OR_ONE_BOOLEAN` (like happens with `cmakedefined01`)